groovy/gradle/src/org/netbeans/modules/gradle/api/ModuleSearchSupport.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.netbeans.modules.gradle.api;
+
+import java.util.Set;
+
+/**
+ *
+ * @author Laszlo Kishalmi
+ */
+public interface ModuleSearchSupport {
+
+    public Set<GradleDependency.ModuleDependency> findModules(String group, String artifact, String version);
+
+    public Set<GradleDependency.ModuleDependency> findModules(String gav);
```

Comment:
A [request/response pattern](http://wiki.apidesign.org/wiki/RequestResponse) might be more extensible:
```java
public final class Query {
  getGav();
  getGroup();
  getArtifact();
  getVersion();
}
public Set<GradleDependency.ModuleDependency> findModules(Query q);
```